### PR TITLE
Set panic as default fallible system param behavior

### PR DESCRIPTION
**Pull request by MiniaczQ**
# Objective

Fixes: #16578

## Solution

This is a patch fix, proper fix requires a breaking change.

Added `Panic` enum variant and using is as the system meta default.
Warn once behavior can be enabled same way disabling panic (originally disabling wans) is.

To fix an issue with the current architecture, where **all** combinator system params get checked together,
combinator systems only check params of the first system.
This will result in old, panicking behavior on subsequent systems and will be fixed in 0.16.

## Testing

Ran unit tests and `fallible_params` example.

### CODE REVIEW

**Comment by MiniaczQ** [comment]
Naive approach comes with some issues, I'm not sure when I'll have time to truly sit down and work on this, so feel free to take over as needed

**Comment by alice-i-cecile** [comment]
Can you say a bit more about the issues you encountered?

**Comment by MiniaczQ** [comment]
The main issue is stuff that was fixed in the `System::run() -> Option<S::Out>` rework (didn't land in 0.15).
In a combination system, like `a.and(b)` or `a.pipe(b)` we're running param checks for *all* systems.
This results in `resource_exists::<MyRes>.and(do_stuff)` failing params, because `do_stuff` doesn't have `MyRes` even though it wouldn't run because of failed condition.
When we change default fail behavior to panic, this causes crashes.

I've had time to think about it since this PR.
We didn't ship a complete feature, so I think we should be fine to downscope it.
Since combination systems are incomplete, let's just ensure the first system (or *the* system, if no combinations are used) works correctly with fallible params.
In 0.16 we can ship the correct param checking behavior, where we do it directly before running a system.

**Comment by alice-i-cecile** [comment]
I'm okay with that as the behavior for now :)

File: crates/bevy_ecs/src/schedule/executor/mod.rs [hunk 1]
```diff
@@ -321,15 +321,11 @@ mod tests {
         schedule.set_executor_kind(executor);
         schedule.add_systems(
             (
-                // Combined systems get skipped together.
-                (|mut commands: Commands| {
-                    commands.insert_resource(R1);
-                })
-                .pipe(|_: In<()>, _: Res<R1>| {}),
```
Comment (MiniaczQ) [inline crates/bevy_ecs/src/schedule/executor/mod.rs:328]: As agreed, this is no longer supported in this scope

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ecs/src/system/combinator.rs [hunk 2]
```diff
@@ -214,7 +214,7 @@ where
     #[inline]
     unsafe fn validate_param_unsafe(&mut self, world: UnsafeWorldCell) -> bool {
         // SAFETY: Delegate to other `System` implementations.
-        unsafe { self.a.validate_param_unsafe(world) && self.b.validate_param_unsafe(world) }
+        unsafe { self.a.validate_param_unsafe(world) }
```
Comment (andriyDev) [inline crates/bevy_ecs/src/system/combinator.rs:217]: Why is this ok? Surely we should check that both params are valid?

File: crates/bevy_ecs/src/system/combinator.rs [hunk 3]
```diff
@@ -433,7 +433,7 @@ where
 
     unsafe fn validate_param_unsafe(&mut self, world: UnsafeWorldCell) -> bool {
         // SAFETY: Delegate to other `System` implementations.
-        unsafe { self.a.validate_param_unsafe(world) && self.b.validate_param_unsafe(world) }
+        unsafe { self.a.validate_param_unsafe(world) }
```
Comment (andriyDev) [inline crates/bevy_ecs/src/system/combinator.rs:436]: Same here

File: crates/bevy_ecs/src/system/combinator.rs (hunk 2 above)
Comment (andriyDev) [reply crates/bevy_ecs/src/system/combinator.rs:217]: Ah I should have read above in the PR discussion, can you just add a to-do or at least a comment explanation why we've decided to do this?

**Review by bushrat011899 — APPROVED** [review]

**Review by BenjaminBrienen — APPROVED** [review]

**Comment by mockersf** [comment]
https://github.com/MiniaczQ/bevy/pull/1 should get the example working again

**Comment by Danielmelody** [comment]
This is actually a breaking change from 0.15.0, as some system with Single<..> query crashes after a patch update to 0.15.1. Better to intro this in 0.16.0,

**Pull request merged**